Updated optional input definition (#599)
nf-core/modules: README.md

- Named file extensions MUST be emitted for ALL output channels e.g. `path "*.txt", emit: txt`.

> - Optional inputs are not currently supported by Nextflow. However, using an empty list (`[]`) instead of a file can be used to work around this issue when combined with conditional staging of the file when necessary (i.e., using a condition around the `script:` block, with one condition containing the command without the optional file, and the other condition containing the command with the optinal file).

Reviewer: ```suggestion
- Optional inputs are not currently supported by Nextflow. However, passing an empty list (`[]`) instead of a file as a module parameter can be used to work around this issue.
```

Reviewer: I got a little confused with the rest of the text.

Reviewer: Don't think we need to go into specifics here? We probably need to have more illustrative examples in a proper docs section on the website.

Author: Also works, can remove for now then
